add support to delay a job and enqueue it N seconds later

orlandobcrra opened the pull request:
add support to delay a job and enqueue it N seconds later.
those jobs are going to have the `delayed` status.
e.g: delay a job 1 hour.
```
queue_job(main_task_path, params, delay=3600)
```

orlandobcrra committed:
add support to delay a job and enqueue it N seconds later

orlandobcrra commented:
Hi @maelorn @sylvinus, please if you could take a look at this PR when you get a chance. thanks.

maelorn commented:
Hello,

Sorry for the delay. Can you tell us a bit more about the specific use case where you would need this ? Maybe it is something we can already handle right now. Thanks.

orlandobcrra commented:
Hello,

For example if we need to send an email 1 hour later from a given action:
Endpoint A is called on our API and we need to send an email with some info we got from this process but not right away, we need to send the email N minutes later.

maelorn commented:
Allright, we don't indeed have anything in that regard. It seems fine to me. Could you just add a small test validating the wanted behaviour please ?

kevteg committed:
getting latest job with certain status

kevteg committed:
Moving method to job methods

kevteg committed:
updating name and version

kevteg committed:
0.9.12 version

kevteg committed:
using query instead of status and path

icardenas committed:
042419IC Add support to delete success or cancel jobs that are already expired

icardenas committed:
042419IC draft test for check test_delete_expired_jobs

icardenas committed:
Update Version 0.9.13

icardenas committed:
Update Version 0.9.14

icardenas committed:
Delete TODO word

icardenas committed:
Merge pull request #2 from apploitech/042419IC_add_remove_expires_jobs

042419IC Remove expired jobs

icardenas committed:
042419IC verify version of python to iterate dictionary

icardenas committed:
042419IC Update version

icardenas committed:
Update utils.py

icardenas committed:
update version

icardenas committed:
042419IC Fix and update version

icardenas committed:
Merge pull request #3 from apploitech/042419IC_fix_dictionary_iteration_version

Verify python version to iterate dictionary

icardenas committed:
Add status maxretries in endpoint '/api/datatables/taskexceptions'

icardenas committed:
Add status maxretries in endpoint '/api/datatables/taskexceptions'

icardenas committed:
Add the status field in the search for exceptions

icardenas committed:
042619IC Update Version

icardenas committed:
minor fix bundle

icardenas committed:
Merge pull request #4 from apploitech/042619IC_add_maxretries_in_taskexceptions

042619IC add maxretries in taskexceptions

Darking360 committed:
Adding get latest jobs based on a query

Darking360 committed:
Adding sort

Darking360 committed:
Increasing MRQ lib version

Darking360 committed:
Merge pull request #5 from apploitech/050319MR_add_latest_jobs_query

Add get latest jobs query

ghabrielv committed:
Add Redis SSL support

ghabrielv committed:
Increasing MRQ lib version

ghabrielv committed:
Fix pypy portable url

ghabrielv committed:
Update dockerfile

ghabrielv committed:
Update dockerfile

ghabrielv committed:
Update library psutil

ghabrielv committed:
Update requirements and dockerfile

ghabrielv committed:
Specific log description content type

ghabrielv committed:
Version 0.9.23 and redis password

ghabrielv committed:
Merge pull request #6 from apploitech/redis-ssl

Add Redis SSL support

ghabrielv committed:
Sentry cleanup (#7)

* Sentry cleanup

* Improve exceptions message

* Improve AbortInterrupt, MaxRetriesInterrupt and MaxConcurrencyInterrupt

ghabrielv committed:
Upgrade psutil to 5.8.0 (#8)

* Update psutil to 5.8.0

* MRQ Custom 0.9.28

orlandobcrra committed:
A3S-1593 - if "retry_current_job" is call outside of an MRQ context, then raise exception. (#9)

Thus, the actual worker performing the execution of this task can use its own error handling. E.g. Lambda function.

ghabrielv committed:
A3S-2518 Remove print line and document the process of generating a new version of MRQ (#12)

* Remove print line
* Update readme.md
* Up version

orlandobcrra committed:
Update processes.py

Commit:
A3S-2202: version upgrade

ghabrielv committed:
A3S-2753: Updated redis to 4.5.5 version

ghabrielv committed:
A3S-2749: Settled MRQ version to 0.9.32

Sarahoyos committed:
Merge pull request #13 from apploitech/A3S-2753

A3S-2753: Updated redis to 4.5.5 version and new MRQ-custom version

ghabrielv committed:
A3S-2753 Fixed redis SSL connection (#14)

* A3S-2753: Fixed redis connection

ghabrielv committed:
A3S-2753: Fixed redis SSL connection (#15)

* A3S-2753: Fixed redis connection
* A3S-2753: Fixed ssl_cert_reqs attribute
* A3S-2753: Improved fix ssl_cert_reqs attribute
* A3S-2793: Updated mrq-custom to version 0.9.34.1

Commit:
A3S-843: update pymongo to support python 3.11

jefarr-apploi committed:
Merge pull request #16 from apploitech/a3s-843_update_pymongo

A3s 843 update pymongo to support python 3.11
